FWCore/Framework/src/EventProcessor.cc

```diff
@@ -2386,6 +2390,19 @@ namespace edm {
     return schedule_->terminate();
   }
 
+  void EventProcessor::tryToMakeOtherStreamsSkipLumi(unsigned int iStreamIndex,
+                                                     LuminosityBlockProcessingStatus& iStatus) {
+    for (unsigned int stream = 0; stream < preallocations_.numberOfStreams(); ++stream) {
+      if (stream == iStreamIndex) {
+        continue;
+      }
+      if (iStatus.setStreamStartedLumi(stream)) {
+        // This cannot be the last stream to end the lumi, but we must decrement the counter
+        iStatus.streamFinishedLumi();
```

Comment:
Would it be worth of asserting the expectation?
```suggestion
        // This cannot be the last stream to end the lumi, but we must decrement the counter
        bool lastStreamToFinish = iStatus.streamFinishedLumi();
        assert(not lastStreamToFinish);
```
Although I'd expect skipping the `iStreamIndex` element, and the `tryToMakeOtherStreamsSkipLumi()` function being called from serial section, to guarantee that.

Reply:
I added the assert and edited the comment also (I'll push it soon with changes related to your other comments).

It cannot be last because the stream associated with ```iStreamIndex``` does not call ```streamEndLumiAsync``` until after ``` tryToMakeOtherStreamsSkipLumi``` returns. Maybe an assert is a good idea. It shouldn't ever fail but if it did and this was the last, then without the assert the Framework would just silently get stuck forever because ```globalEndLumiAsync``` would never get called.